### PR TITLE
fix(Notification): fix notification overflow on small screens

### PR DESCRIPTION
**Pull request by vvmarulin**
[TACO-1516]

### Description

On small screens, notifications overflow the screen. That makes it impossible to read a notification message or close it with `x` icon. That happens because we fixed the width of `SnackbarContent` to `27.5em` for info notification type and to `20em` for other notification types. 

The solution is to add `max-width: 100%` to `SnackbarContent` styles. But that property is ignored by the browser because the parent element has `width: auto` which is set by the `notistack` package. That was fixed in [version `1.0.6`](https://github.com/iamhosseindhv/notistack/commit/fe4b7c8edd53189a9f46b3032e276aaef98f76b2). That's why I have to upgrade version of `notistack` from `1.0.5` to `1.0.6`.

### How to reproduce the bug

- Open https://picasso.toptal.net/?path=/story/components-notification--notification#notifications-stream-default
- Change the size of the preview to `Small mobile`
- Click "Show general notification"
- The notification will overflow the preview and you won't be able to see "x" icon

### How to test

- Open https://picasso.toptal.net/fix-use-notification-overflow/?path=/story/components-notification--notification#notifications-stream-default
- Change the size of the preview to `Small mobile`
- Click "Show general notification"
- Notification should fit in the preview this time.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|  image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[TACO-1516]: https://toptal-core.atlassian.net/browse/TACO-1516?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/large-badgers-remain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso-provider': patch
+---
+
+Upgrade `notistack` to version `1.0.6`
```
Comment (LashaJini) [inline .changeset/large-badgers-remain.md:5]: I don't think it's necessary to have this (and the same one in another changeset) info in changeset

File: .changeset/wicked-cycles-kneel.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+'@toptal/picasso': patch
+---
+
+---
+
+# Notification
```
Comment (LashaJini) [inline .changeset/wicked-cycles-kneel.md:7]:
```suggestion
### Notification
```

File: .changeset/wicked-cycles-kneel.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+Fix overflow of notifications on the small screens
```
Comment (LashaJini) [inline .changeset/wicked-cycles-kneel.md:9]:
```suggestion
- Fix overflow of notifications on the small screens
```

**Review by LashaJini — APPROVED** [review]
Tested locally. Works gr8. Thank you for fix :+1:

File: .changeset/large-badgers-remain.md (hunk 1 above)
Comment (vvmarulin) [reply .changeset/large-badgers-remain.md:5]: `notistack` is a peer dependency in `@toptal/picasso`. So it has to be updated in the dependent projects. That's why I add updates to `picasso-provider` and `picasso-shared`.

Or do you mean that we need a different message in the changeset?

Comment (LashaJini) [reply .changeset/large-badgers-remain.md:5]: Ah, got it, missed that part. :+1:

Comment (OndrejTuma) [reply .changeset/large-badgers-remain.md:5]: Just to be clear - and correct me if I'm wrong - we could possibly squash those two changesets together right?

Making:
```
---
'@toptal/picasso-provider': patch
'@toptal/picasso-shared': patch
---

Upgrade `notistack` to version `1.0.6`
```

**Review by OndrejTuma — APPROVED** [review]
Tested in temploy, behaves correctly 👍 
There is incorrect align of error notification on `Large mobile` but thats above scope of this PR

**Review by ozgurkececioglu — APPROVED** [review]
Looking good 👍 Thanks for the contribution 🙇‍♂️

**Review by separatio — APPROVED** [review]

**Pull request merged**